pkg/providers/tinkerbell/controlplane.go

```diff
@@ -51,6 +55,23 @@ func ControlPlaneSpec(ctx context.Context, logger logr.Logger, client kubernetes
 		return nil, errors.Wrap(err, "generating tinkerbell control plane yaml spec")
 	}
 
+	capiCluster, err := controller.GetCAPICluster(ctx, client, clusterSpec.Cluster)
+	if err != nil {
+		return nil, err
+	}
+	if capiCluster != nil {
+		rollingOverScale, err := anywherev1.IsRollingNotScale(ctx, client, clusterSpec.Cluster)
```

Comment:
Where is the comparison happening?

Reply:
please find detailed comparator here. https://github.com/aws/eks-anywhere/pull/4994